Exception on `GET /cultural_heritage_item` request

I got this exception on request:
```
GET /cultural_heritage_item
Authorization: JWT <token>
```
Reason is probably the optimistic regex match on `cultural_heritage_item` endpoint. From the client side, I expect three endpoints:

- `GET /cultural_heritage_items`: Get all cultural heritage items
- `GET /cultural_heritage_items/:id`: Get cultural heritage item with id :id
- `POST /cultural_heritage_items`: Create a cultural heritage item
```
Internal Server Error: /cultural_heritage_item
Traceback (most recent call last):
  File "/Users/yigitozkavci/Documents/Programming/PythonWorkspace/bounswe2017group10/Atlas/backend/atlasenv/lib/python3.6/site-packages/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "/Users/yigitozkavci/Documents/Programming/PythonWorkspace/bounswe2017group10/Atlas/backend/atlasenv/lib/python3.6/site-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/Users/yigitozkavci/Documents/Programming/PythonWorkspace/bounswe2017group10/Atlas/backend/atlasenv/lib/python3.6/site-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/yigitozkavci/Documents/Programming/PythonWorkspace/bounswe2017group10/Atlas/backend/atlasenv/lib/python3.6/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/Users/yigitozkavci/Documents/Programming/PythonWorkspace/bounswe2017group10/Atlas/backend/atlasenv/lib/python3.6/site-packages/django/views/generic/base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "/Users/yigitozkavci/Documents/Programming/PythonWorkspace/bounswe2017group10/Atlas/backend/atlasenv/lib/python3.6/site-packages/rest_framework/views.py", line 489, in dispatch
    response = self.handle_exception(exc)
  File "/Users/yigitozkavci/Documents/Programming/PythonWorkspace/bounswe2017group10/Atlas/backend/atlasenv/lib/python3.6/site-packages/rest_framework/views.py", line 449, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/Users/yigitozkavci/Documents/Programming/PythonWorkspace/bounswe2017group10/Atlas/backend/atlasenv/lib/python3.6/site-packages/rest_framework/views.py", line 486, in dispatch
    response = handler(request, *args, **kwargs)
  File "/Users/yigitozkavci/Documents/Programming/PythonWorkspace/bounswe2017group10/Atlas/backend/atlasenv/lib/python3.6/site-packages/rest_framework/generics.py", line 241, in get
    return self.list(request, *args, **kwargs)
  File "/Users/yigitozkavci/Documents/Programming/PythonWorkspace/bounswe2017group10/Atlas/backend/atlasenv/lib/python3.6/site-packages/rest_framework/mixins.py", line 48, in list
    return Response(serializer.data)
  File "/Users/yigitozkavci/Documents/Programming/PythonWorkspace/bounswe2017group10/Atlas/backend/atlasenv/lib/python3.6/site-packages/rest_framework/serializers.py", line 738, in data
    ret = super(ListSerializer, self).data
  File "/Users/yigitozkavci/Documents/Programming/PythonWorkspace/bounswe2017group10/Atlas/backend/atlasenv/lib/python3.6/site-packages/rest_framework/serializers.py", line 262, in data
    self._data = self.to_representation(self.instance)
  File "/Users/yigitozkavci/Documents/Programming/PythonWorkspace/bounswe2017group10/Atlas/backend/atlasenv/lib/python3.6/site-packages/rest_framework/serializers.py", line 656, in to_representation
    self.child.to_representation(item) for item in iterable
  File "/Users/yigitozkavci/Documents/Programming/PythonWorkspace/bounswe2017group10/Atlas/backend/atlasenv/lib/python3.6/site-packages/rest_framework/serializers.py", line 656, in <listcomp>
    self.child.to_representation(item) for item in iterable
  File "/Users/yigitozkavci/Documents/Programming/PythonWorkspace/bounswe2017group10/Atlas/backend/atlasenv/lib/python3.6/site-packages/rest_framework/serializers.py", line 483, in to_representation
    fields = self._readable_fields
  File "/Users/yigitozkavci/Documents/Programming/PythonWorkspace/bounswe2017group10/Atlas/backend/atlasenv/lib/python3.6/site-packages/django/utils/functional.py", line 35, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/Users/yigitozkavci/Documents/Programming/PythonWorkspace/bounswe2017group10/Atlas/backend/atlasenv/lib/python3.6/site-packages/rest_framework/serializers.py", line 377, in _readable_fields
    field for field in self.fields.values()
  File "/Users/yigitozkavci/Documents/Programming/PythonWorkspace/bounswe2017group10/Atlas/backend/atlasenv/lib/python3.6/site-packages/rest_framework/serializers.py", line 363, in fields
    for key, value in self.get_fields().items():
  File "/Users/yigitozkavci/Documents/Programming/PythonWorkspace/bounswe2017group10/Atlas/backend/atlasenv/lib/python3.6/site-packages/rest_framework/serializers.py", line 993, in get_fields
    field_names = self.get_field_names(declared_fields, info)
  File "/Users/yigitozkavci/Documents/Programming/PythonWorkspace/bounswe2017group10/Atlas/backend/atlasenv/lib/python3.6/site-packages/rest_framework/serializers.py", line 1070, in get_field_names
    serializer_class=self.__class__.__name__
AssertionError: ("Creating a ModelSerializer without either the 'fields' attribute or the 'exclude' attribute has been deprecated since 3.3.0, and is now disallowed. Add an explicit fields = '__all__' to the cultural_heritage_serializer serializer.",)
[18/Oct/2017 11:58:02] "GET /cultural_heritage_item HTTP/1.1" 500 22281
```

--- a/Atlas/backend/atlas/serializers.py
+++ b/Atlas/backend/atlas/serializers.py
@@ -9,3 +9,4 @@ class cultural_heritage_serializer(serializers.ModelSerializer):
     user = AccountSerializer
     class Meta:
         model = Cultural_Heritage
+        fields ='__all__'

--- a/Atlas/backend/atlas/tests.py
+++ b/Atlas/backend/atlas/tests.py
@@ -180,3 +180,22 @@ class cultural_heritage_item(TestCase):
         self.assertEqual(response_content['title'],title)
         self.assertEqual(response_content['id'],id)
 
+
+    def test_get_cultural_heritage_item_by_invalid_id(self):
+        title = 'Very emotional thresh hook'
+        item_data = {
+            "title": title,
+        }
+        response = self.client.post(
+            self.cultural_heritage_item_url,
+            item_data,
+            format='json',
+
+        )
+        self.assertEqual(response.status_code, 201)
+        response_content = json.loads(smart_text(response.content))
+        response = self.client.get(
+            self.cultural_heritage_item_url + '12321321',
+            format='json',
+        )
+        self.assertEqual(response.status_code, 404)

--- a/Atlas/backend/atlas/tests.py
+++ b/Atlas/backend/atlas/tests.py
@@ -112,3 +112,71 @@ class cultural_heritage_item(TestCase):
 
         )
         self.assertEqual(response.status_code, 400)
+
+    def test_get_cultural_heritage_item_by_id(self):
+        title = 'Very emotional thresh hook'
+        item_data = {
+            "title": title,
+        }
+        response = self.client.post(
+            self.cultural_heritage_item_url,
+            item_data,
+            format='json',
+
+        )
+        self.assertEqual(response.status_code, 201)
+        response_content = json.loads(smart_text(response.content))
+        id =response_content['id']
+        response = self.client.get(
+            self.cultural_heritage_item_url + str(id),
+            format='json',
+        )
+        self.assertEqual(response.status_code,200)
+        response_content = json.loads(smart_text(response.content))
+        self.assertEqual(response_content['title'],title)
+        self.assertEqual(response_content['id'],id)
+
+    def test_get_cultural_heritage_item_by_id_with_guest_user(self):
+        title = 'Very emotional thresh hook'
+        item_data = {
+            "title": title,
+        }
+        response = self.client.post(
+            self.cultural_heritage_item_url,
+            item_data,
+            format='json',
+
+        )
+        self.assertEqual(response.status_code, 201)
+        response_content = json.loads(smart_text(response.content))
+        id = response_content['id']
+        self.client.logout()
+        response = self.client.get(
+            self.cultural_heritage_item_url + str(id),
+            format='json',
+        )
+        self.assertEqual(response.status_code, 403)
+
+    def test_get_cultural_heritage_item_by_id_regex(self):
+        title = 'Very emotional thresh hook'
+        item_data = {
+            "title": title,
+        }
+        response = self.client.post(
+            self.cultural_heritage_item_url,
+            item_data,
+            format='json',
+
+        )
+        self.assertEqual(response.status_code, 201)
+        response_content = json.loads(smart_text(response.content))
+        id = response_content['id']
+        response = self.client.get(
+            self.cultural_heritage_item_url + str(id)+'/',
+            format='json',
+        )
+        self.assertEqual(response.status_code, 200 )
+        response_content = json.loads(smart_text(response.content))
+        self.assertEqual(response_content['title'],title)
+        self.assertEqual(response_content['id'],id)
+

--- a/Atlas/backend/atlas/urls.py
+++ b/Atlas/backend/atlas/urls.py
@@ -29,6 +29,7 @@ urlpatterns = [
     url(r'^users/', views.users, name='users'),
     url(r'^api/auth/', include('authentication.urls')),
     url(r'^cultural_heritage_item/?$', views.cultural_heritage_item.as_view()),
+    url(r'^cultural_heritage_item/(?P<id>\d+)/?$', views.cultural_heritage_item_view_update_delete.as_view()),
 ]
 
 urlpatterns += staticfiles_urlpatterns()

--- a/Atlas/backend/atlas/views.py
+++ b/Atlas/backend/atlas/views.py
@@ -24,3 +24,13 @@ class cultural_heritage_item(generics.ListCreateAPIView):
         #Pk is the id of the user.
         request.data['user'] = request.user.pk
         return super(cultural_heritage_item, self).create(request,*args,**kwargs)
+
+class cultural_heritage_item_view_update_delete(generics.RetrieveUpdateDestroyAPIView):
+    serializer_class = cultural_heritage_serializer
+    lookup_field = 'id'
+    def get_queryset(self):
+        return Cultural_Heritage.objects.filter()
+
+
+
+

